app/domain/services/fetch_assignment_spes/service.rb

```diff
@@ -45,6 +49,8 @@ def process(spe_requests:)
 
         {
           request_uuid: request_uuid,
+          calculation_uuid: calculation_uuid,
+          ecosystem_matrix_uuid: ecosystem_matrix_uuid,
           assignment_uuid: assignment_uuid,
           exercise_uuids: exercise_uuids,
           assignment_status: assignment_status,
```

Comment:
This might be a bit clearer if you set the return hash above the if and then set it's properties as needed.  That way you wouldn't need to set the `nil` values since that'd be the default  Something like:
```ruby
response = {}
if assignment_spe.nil?
  response.merge!(
    exercise_uuids: []
    assignment: missing_spe_assignments_by_uuid[assignment_uuid]
  )
else
   response.merge!(
      calculation_uuid: assignment_spe.calculation_uuid,
      ecosystem_matrix_uuid: assignment_spe.ecosystem_matrix_uuid
  ) 
end
response
```

probably not worth updating though